Çeşitli ülkelerin belirli kategorilerdeki verilerini içeren veri setinin analizi yapılacaktır. Öncelikle gerekli kütüphaneler eklenmeli ve veri seti okunmalıdır.

In [1]:
import pandas as pd
df=pd.read_csv("C:\\Users\\beyza\\OneDrive\\Masaüstü\\VeriBilimi\\country.csv")

Ülkeleri nüfusa göre azalan sayıda sıralama:

In [2]:
print("Nüfusa göre azalan sırada ülkeler:\n ", df.sort_values("Population",ascending=False)[["Country", "Population"]])

Nüfusa göre azalan sırada ülkeler:
                     Country  Population
42                  China   1313973713
94                  India   1095351995
214         United States    298444215
95              Indonesia    245452739
27                 Brazil    188078227
..                     ...         ...
144                 Nauru        13287
209                Tuvalu        11810
140            Montserrat         9439
171          Saint Helena         7502
174  St Pierre & Miquelon         7026

[227 rows x 2 columns]


En yüksek nüfusa sahip ülke 1.313.973.713 ile Çin, en düşük ise 7026 ile St Pierre & Miquelon çıkmıştır.

Kişi başına düşen ortalama gelire göre ülkeleri artan sayıda sıralama:

In [3]:
print("GDP per capita sütununa göre artan sırada ülkeler:\n ", df.sort_values("GDP ($ per capita)",ascending=True)[["Country", "GDP ($ per capita)"]])

GDP per capita sütununa göre artan sırada ülkeler:
               Country  GDP ($ per capita)
58       East Timor                500.0
183    Sierra Leone                500.0
188         Somalia                500.0
33          Burundi                600.0
125          Malawi                600.0
..               ...                 ...
22          Bermuda              36000.0
154          Norway              37800.0
214   United States              37800.0
121      Luxembourg              55100.0
223  Western Sahara                  NaN

[227 rows x 2 columns]


En düşük ortalama gelir 500 ile East Timor iken en yüksek 55.100 ile Lüksemburgdur. Western Sahara'nın ise GDP bilgisi yoktur.

Nüfusu 10 milyonun üzerinde olan ülkeler:

In [4]:
print("Nüfusu 10 milyonun üzerinde olan ülkeler: \n ", df[df["Population"]>10000000][["Country", "Population"]])

Nüfusu 10 milyonun üzerinde olan ülkeler: 
            Country  Population
0    Afghanistan     31056997
2        Algeria     32930091
5         Angola     12127071
8      Argentina     39921833
11     Australia     20264082
..            ...         ...
218    Venezuela     25730435
219      Vietnam     84402966
224        Yemen     21456188
225       Zambia     11502010
226     Zimbabwe     12236805

[79 rows x 2 columns]


Nüfusu 10 milyonun üzerinde olan 79 ülke vardır.

Okur yazarlığa göre ülkeleri sıralayıp en yüksek okur-yazarlık oranına sahip ilk 5 ülkeyi seçme:

In [5]:
print("En yüksek okur yazarlığa sahip 5 ülke: \n ", df.sort_values("Literacy (%)",ascending=False)[["Country", "Literacy (%)"]].head(5))

En yüksek okur yazarlığa sahip 5 ülke: 
               Country Literacy (%)
53   Czech Republic          99,9
93          Iceland          99,9
163          Poland          99,8
114          Latvia          99,8
64          Estonia          99,8


En yüksek okur yazarlığa %99,9 ile Çek Cumhuriyeti sahiptir.

Kişi başına düşen ortalama geliri 10.000'in üzerinde olan ülkeleri filtreleme:


In [6]:
filtered_df=df[df["GDP ($ per capita)"]>10000]
print("Kişi Başı GSYİH 10.000'in üzerinde olan ülkeler:\n ", filtered_df[["Country","GDP ($ per capita)"]])

Kişi Başı GSYİH 10.000'in üzerinde olan ülkeler:
                     Country  GDP ($ per capita)
4                 Andorra              19000.0
7       Antigua & Barbuda              11000.0
8               Argentina              11200.0
10                  Aruba              28000.0
11              Australia              29000.0
..                     ...                 ...
212  United Arab Emirates              23200.0
213        United Kingdom              27700.0
214         United States              37800.0
215               Uruguay              12800.0
220        Virgin Islands              17200.0

[76 rows x 2 columns]


Kişi Başı GSYİH 10.000$'ın üzerinde olan 76 ülke vardır.

Nüfus yoğunluğuna göre ülkeleri sıralayıp en yüksek nüfus yoğunluğuna sahip ilk 10 ülkeyi seçme:

In [7]:
print("En yüksek nüfus yoğunluğuna sahip 10 ülke:\n", df.sort_values("Pop. Density (per sq. mi.)",ascending=False)[["Country","Pop. Density (per sq. mi.)"]].head(10))

En yüksek nüfus yoğunluğuna sahip 10 ülke:
          Country Pop. Density (per sq. mi.)
9       Armenia                        99,9
186    Slovenia                        99,2
202        Togo                        97,7
12      Austria                        97,7
168     Romania                        93,9
77        Ghana                        93,6
140  Montserrat                        92,5
13   Azerbaijan                        91,9
56     Dominica                        91,4
206      Turkey                        90,2


En yüksek nüfus yoğunluğuna sahip ülke %99,9 ile Ermenistan. 10. ise %90,2 ile Türkiye'dir.